dmscripts/notify_buyers_to_award_closed_briefs.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, date, timedelta
+
+import dmapiclient
+from dmutils.email import DMNotifyClient
+from dmutils.email.exceptions import EmailError
+from dmutils.formats import DATE_FORMAT
+
+from dmscripts.helpers import logging_helpers, date_helpers, brief_data_helpers
+
+logger = logging_helpers.configure_logger({'dmapiclient': logging_helpers.logging.INFO})
+
+
+def _filter_briefs_by_user_id_list(briefs, user_id_list):
+    """Return a list of briefs if they have at least 1 user in the given ID list"""
+    if user_id_list:
+        return [brief for brief in briefs if any(filter(lambda user: user['id'] in user_id_list, brief['users']))]
+    return briefs
+
+
+def _create_context_for_brief(brief):
+    return {
+        'brief_id': brief['id'],
+        'brief_title': brief['title'],
+        'framework_slug': brief['frameworkSlug'],
+        'lot_slug': brief['lotSlug'],
+        'utm_date': date.today().strftime("%Y%m%d")
+    }
+
+
+def _get_brief_closing_date(offset_days, closing_date_arg=None):
+    """Returns the closing date object we will query briefs by.
+    If no <closing_date_arg> is supplied, use '<offset_days> ago' as the date.
+
+    If the <closing_date_arg> string is supplied, then check it's not less than <offset_days> ago
+     (otherwise we'll email the buyers too soon after the closing date).
+     e.g. On the 29th Jan, closing_date_arg=2016-01-01 offset_days=28 will return date(2016, 1, 1)
+          On the 28th Jan, closing_date_arg=2016-01-01 offset_days=28 will return False
+    """
+    if closing_date_arg:
+        closing_date = date_helpers.get_date_closed(closing_date_arg)
+        if closing_date > (datetime.utcnow() - timedelta(days=offset_days)).date():
+            return False
+        return closing_date
+    return date.today() - timedelta(days=offset_days)
+
+
+def send_email_to_brief_user_via_notify(notify_client, notify_template_id, user, brief, user_id_list, dry_run):
+    if user_id_list and user['id'] not in user_id_list:
+        # If a user ID list is supplied, only email users for this brief that are in the list
+        return
+
+    logging_context = {
+        'brief_title': brief['title'],
+        'brief_id': brief['id'],
+        'no_of_users': len(brief['users']),
+    }
+    email_context_data = _create_context_for_brief(brief)
+    if user['active']:
+        try:
+            if dry_run:
+                logger.info(
+                    "Would notify {no_of_users} user(s) about brief ID: {brief_id} - '{brief_title}'",
```

Comment:
The log messages here don't seem quite right now - these are going to be output per-user rather than per-brief, so probably should be something more like `"Would notify {email_address} about brief ID: {brief_id} - '{brief_title}'`

For the dry run we can output actual email addresses as it helps with debugging, and perhaps we don't need to log each email for the non-dry-run case?  (That's how we do it here: https://github.com/alphagov/digitalmarketplace-scripts/blob/master/dmscripts/upload_counterpart_agreements.py#L83)

Or maybe for the non-dry-run we log the headline number somewhere before the call to `send_email_to_brief_user_via_notify`?

Reply:
(Either way we probably don't want actual email addresses output in live logs)

Reply:
Good spot. How about logging user IDs instead of emails though (like we do with errors)? Still easy enough to debug but no personal info logged on live?

Reply:
The dry run now looks something like this:

```
2017-10-09 13:55:27,636 script INFO Notifying users about 1 closed brief(s)
2017-10-09 13:55:27,636 script INFO Would notify user ID 34321 about brief ID 5148: 'Digital Discovery: Exploring the Potential for Business Energy Efficiency Website'
```

And the wet run would be (including the Notify client's logging):

```
2017-10-09 13:55:27,636 script INFO Notifying users about 1 closed brief(s)
2017-10-09 13:55:27,636 script INFO Notifying user ID 34321 about brief ID 5148: 'Digital Discovery: Exploring the Potential for Business Energy Efficiency Website'
2017-10-09 13:55:27,636 script INFO Sent email M7o-Z5TLAM6McxNgHy0tk-aAnkzXB6vKeLh0AzLHuwA= to xID-VUdAEenUtL2p_b8rLRiUO4eRYxV1WK7TNDI1zUc= (id: 75216ef4-e42a-4751-ac46-a49887f71cca, template: bc1f93a8-9529-48b6-9a82-af61bc6da332) through Notify
```

Reply:
Yep this seems good to me.